GeneratorInterface/Core/interface/ConcurrentGeneratorFilter.h

```diff
@@ -299,24 +315,39 @@ namespace edm {
   }
 
   template <class HAD, class DEC>
-  void ConcurrentGeneratorFilter<HAD, DEC>::globalBeginLuminosityBlockProduce(LuminosityBlock& lumi,
-                                                                              EventSetup const& es) const {
-    // need one of the streams to finish
+  std::shared_ptr<gen::GenLumiCache<HAD, DEC>> ConcurrentGeneratorFilter<HAD, DEC>::globalBeginLuminosityBlock(
+      edm::LuminosityBlock const&, edm::EventSetup const&) const {
+    //need one of the streams to finish
     while (useInLumi_.load() == nullptr) {
     }
+
```

Comment:
How about a comment here explaining why the extra wait?

Reply:
Done. Thanks.